DDF-5767 Fix 2D map labels disappearing in subsequent searches and clustering

rfding opened the pull request:
### Original DDF PR: codice/ddf#5768 (4 approvals)

#### What does this PR do?
Fixes 2D map labels

#### Who is reviewing it? 
<!--(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)-->
@mdang8 
@josephthweatt 
@aaronilovici 

#### Select relevant component teams: 
<!--
@codice/build 
@codice/continuous-integration 
@codice/core-apis 
@codice/data 
@codice/docs 
@codice/io 
@codice/ogc 
@codice/security 
@codice/solr 
@codice/test 
@codice/ui 
@codice/website 
-->
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
<!--
If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below)
@adimka
@ahoffer
@andrewkfiedler
@AzGoalie
@bdeining
@bdthomson
@blen-desta
@brendan-hofmann
@brjeter
@clockard
@coyotesqrl
@djblue
@emmberk
@figliold
@garrettfreibott
@gordocanchola 
@jlcsmith
@jrnorth
@lambeaux
@lessarderic
@mcalcote
@millerw8
@oconnormi
@paouelle
@pklinef
@ricklarsen - Documentation
@ryeats
@rzwiefel
@shaundmorris
@stustison
@tbatie
@troymohl
@vinamartin
-->
@vinamartin 
@andrewkfiedler 

#### How should this be tested?
<!--(List steps with links to updated documentation)-->
1. Open the Admin console and navigate to `Search UI -> Configuration -> Catalog UI Search`
2. Scroll down to `Show Custom Text Notations` and click the checkbox to enable it
3. Ingest some metacards in Intrigue (make sure some overlap)
4. Open the 2D map and run a * search
5. Verify the metacard map markers have a label next to each one
6. Rerun the search
7. Verify the map labels are still there
8. Enable clustering
9. Verify the non-clustered map markers have labels

Also verify that there are never any labels in the same location overlapping when selecting metacards, especially after re-running the search.

#### Any background context you want to provide?
Fixes functionality introduced in codice/ddf#5638 
The fix for the bug removed the benefit of 2D map labels not overlapping when zoomed out very far from the map. I also had to manually implement hiding logic for overlapping labels, but I think it is more important that the map labels continue to show up in subsequent searches. If there are any better ways to do this, please let me know.

#### What are the relevant tickets?
Fixes: codice/ddf#5767 

#### Screenshots
<!--(if appropriate)-->

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

pklinef commented:
@stustison and @andrewkfiedler might want to also review since I updated lodash.

jlcsmith commented:
Reviewed and tested - looks good to me.

pklinef commented on search-ui/standard/src/main/webapp/main.js:
```diff
@@ -173,9 +173,17 @@ require(['jquery',
+    function (_, $, Backbone, Marionette, app, ich, properties) {
         'use strict';
 
+        // Make lodash compatible with Backbone
```
This is based on https://github.com/lodash/lodash/blob/3.7.0/test/backbone.html#L33-L42

andrew-fiedler commented:
Checked it out, looks good.

Commit:
Selected label displays when labels overlap on 3D map

Commit:
Changed 2D map labels to be on different vector layers
This fixes the problem of labels disappearing in new searches and when clustering is turned on.

Commit:
Removed unknown variable

Commit:
Factored out label show/hide logic

Commit:
Simplified label show/hide logic

Commit:
Remove labels from map model when geometry is destroyed

Commit:
Remove label deletes one label instead of all labels

Commit:
Added missing require

Commit:
Fixed incorrect require
